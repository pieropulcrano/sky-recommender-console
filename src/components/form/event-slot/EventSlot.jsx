import React from 'react';
import PropTypes from 'prop-types';
import { useField, useFormikContext } from 'formik';
import { formatToHumanReadable } from '../../../utils/date';
import IconButton from '@mui/material/IconButton';
import ClearIcon from '@mui/icons-material/Clear';
import Grid from '@mui/material/Grid';
import { Tooltip } from '@material-ui/core';
import { ErrorMsg } from '../error-msg/ErrorMsg.styled';
import {
  SlotWrapper,
  EventImageWrapper,
  XButton,
  EventImage,
  EmptyEventWrapper,
  HD,
  SD,
  Warning,
  ImageNotAvailable,
  EventTitle,
  EventDateTime,
  AddIcon,
  StartDateIcon,
  EndDateIcon,
  EventChannel,
} from './EventSlot.styled';

/**
 * Component to display an assigned event or select a new one, connected with Formik state by the param "name"
 */

const SLOT_TYPE_HD = 'hd';
const SLOT_TYPE_SD = 'sd';

const EventSlot = ({ name, handleOpen, type, disabled, data_test_slot }) => {
  const [field, meta] = useField(name);
  const { setFieldValue } = useFormikContext();
  const { value } = field;

  const rmvEvent = () => setFieldValue(name, {});

  const handleOpenModal = () => handleOpen(name);

  if (!value || Object.keys(value).length === 0) {
    return (
      <SlotWrapper className="empity-slot">
        <EventImageWrapper error={meta && meta.touched && meta.error}>
          {type === SLOT_TYPE_SD ? <SD /> : type === SLOT_TYPE_HD && <HD />}
          <EmptyEventWrapper error={meta && meta.touched && meta.error}>
            {!disabled && (
              <IconButton onClick={handleOpenModal}>
                <AddIcon fontSize="large" />
              </IconButton>
            )}
          </EmptyEventWrapper>
        </EventImageWrapper>
        {meta && meta.touched && meta.error && (
          <ErrorMsg>This slot is required to continue saving</ErrorMsg>
        )}
      </SlotWrapper>
    );
  }

  const renderTooltipWarning = (message) => {
    const wariningMessages = message.split(/\r?\n/);
    const listItems = wariningMessages.map((m, idx) => {
      if (m !== '') {
        return <li key={idx}>{m.trim()}</li>;
      }
    });
    return listItems;
  };

  return (
    <SlotWrapper data-test-slot={data_test_slot} data-test={value.id}>
      <EventImageWrapper>
        {type === SLOT_TYPE_SD ? <SD /> : type === SLOT_TYPE_HD && <HD />}
        {!disabled && (
          <XButton onClick={rmvEvent}>
            <ClearIcon color="error" fontSize="small" />
          </XButton>
        )}
        {value.warningMessage && (
          <Tooltip
            data-testid="warningIcon"
            title={renderTooltipWarning(value.warningMessage)}
          >
            <Warning />
          </Tooltip>
        )}
        {(value.verticalImageUrl && (
          <EventImage src={value.verticalImageUrl} />
        )) || (
          <EmptyEventWrapper>
            <ImageNotAvailable />
          </EmptyEventWrapper>
        )}
      </EventImageWrapper>
      {value.title && (
        <Grid container>
          <Tooltip title={value.title}>
            <EventTitle data-test="event-title" noWrap>
              {value.title}
            </EventTitle>
          </Tooltip>
        </Grid>
      )}
      {value.channelName && value.type === 'LIN' && (
        <Grid container>
          <Tooltip title={value.channelName}>
            <EventChannel noWrap>{value.channelName}</EventChannel>
          </Tooltip>
        </Grid>
      )}
      {value.startProgram && (
        <Grid container justify="flex-end" alignItems="flex-end">
          <Tooltip title="Start Program">
            <StartDateIcon />
          </Tooltip>
          <EventDateTime data-test="event-startProgram" noWrap>
            {value.startProgram && formatToHumanReadable(value.startProgram)}
          </EventDateTime>
        </Grid>
      )}
      {value.endProgram && (
        <Grid container justify="flex-end" alignItems="flex-end">
          <Tooltip title="End Program">
            <EndDateIcon />
          </Tooltip>
          <EventDateTime data-test="event-endProgram" noWrap>
            {value.endProgram && formatToHumanReadable(value.endProgram)}
          </EventDateTime>
        </Grid>
      )}
    </SlotWrapper>
  );
};

EventSlot.defaultProps = {
  type: undefined,
  disabled: false,
};

EventSlot.propTypes = {
  /**
   * The field name that connects the table with the Formik state.
   */
  name: PropTypes.string.isRequired,
  /**
   * Gets called when the user click on the add button.
   */
  handleOpen: PropTypes.func.isRequired,
  /**
   * Indicates the type of the event assigned to the slot.
   */
  type: PropTypes.oneOf([SLOT_TYPE_SD, SLOT_TYPE_HD]),
  /**
   * Disable all possible interaction with the component (add / remove event).
   */
  disabled: PropTypes.bool,
};

export default EventSlot;
