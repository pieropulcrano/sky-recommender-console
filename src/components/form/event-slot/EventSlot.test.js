import React from 'react';
import * as Yup from 'yup';
import { Formik } from 'formik';
import userEvent from '@testing-library/user-event';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import EventSlot from './EventSlot';

describe('Event Slot', () => {
  let props;
  let initialValues;
  let validationSchema;

  beforeEach(() => {
    props = {
      name: 'eventSlot',
      handleOpen: jest.fn(),
      options: {
        opt_1: 'opt-1',
        opt_2: 'opt-2',
      },
    };
    initialValues = { eventSlot: {} };
    validationSchema = Yup.object().shape({
      eventSlot: Yup.object().shape({
        id: Yup.string().required(),
        position: Yup.string().required(),
        title: Yup.string().required(),
      }),
    });
  });

  describe('empty', () => {
    it('should render', () => {
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render a disabled slot', () => {
      props.disabled = true;
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render a disabled plus slot', () => {
      props.disableAddIcon = true;
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render an hd slot', () => {
      props.type = 'hd';
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render an sd slot', () => {
      props.type = 'sd';
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render an hd disabled slot', () => {
      props.type = 'hd';
      props.disabled = true;
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should call handleOpen callback', async () => {
      const { container } = render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      const addButton = container.querySelector('button');
      await fireEvent.click(addButton);

      expect(props.handleOpen).toHaveBeenCalledTimes(1);
    });
  });

  describe('not empty', () => {
    beforeEach(() => {
      initialValues = {
        eventSlot: {
          title: 'event-title',
          startProgram: '2021-12-24T12:00:00Z',
          endProgram: '2022-12-24T12:00:00Z',
          warningMessage: 'out of date',
        },
      };
    });

    it('should render', () => {
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render a disabled slot', () => {
      props.disabled = true;

      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should render an hd slot', () => {
      props.type = 'hd';

      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should remove the assigned event', async () => {
      const { container } = render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      const removeButton = container.querySelector('button');
      await waitFor(() => fireEvent.click(removeButton));

      expect(document.body.childNodes[0].children).toMatchSnapshot();
    });

    it('should display the warning tooltip', async () => {
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );
      let warningIcon = screen.getByTestId('warningIcon');
      await userEvent.hover(warningIcon);
      const tooltipText = await screen.findByText('out of date');

      expect(tooltipText).toBeInTheDocument();
    });

    it('should display the start date tooltip', async () => {
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      let startDate = screen.getByTestId(/EventAvailableIcon/);
      await userEvent.hover(startDate);
      const tooltipText = await screen.findByText(/Start Program/);

      expect(tooltipText).toBeInTheDocument();
    });

    it('should display the end date tooltip', async () => {
      render(
        <Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
        >
          <EventSlot {...props} />
        </Formik>,
      );

      let endDate = screen.getByTestId(/EventBusyIcon/);
      await userEvent.hover(endDate);
      const tooltipText = await screen.findByText(/End Program/);

      expect(tooltipText).toBeInTheDocument();
    });
  });
});
