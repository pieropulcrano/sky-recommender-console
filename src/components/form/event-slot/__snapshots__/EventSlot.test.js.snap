// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Slot empty should render 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1fyogec-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Event Slot empty should render a disabled plus slot 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <button
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-9dlxcj-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Event Slot empty should render a disabled slot 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <div
        class="MuiBox-root css-1vwnyhv"
      />
    </div>
  </div>,
]
`;

exports[`Event Slot empty should render an hd disabled slot 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root css-1kosrk6"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-8 12H9.5v-2h-2v2H6V9h1.5v2.5h2V9H11v6zm2-6h4c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1h-4V9zm1.5 4.5h2v-3h-2v3z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      />
    </div>
  </div>,
]
`;

exports[`Event Slot empty should render an hd slot 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root css-1kosrk6"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-8 12H9.5v-2h-2v2H6V9h1.5v2.5h2V9H11v6zm2-6h4c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1h-4V9zm1.5 4.5h2v-3h-2v3z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1fyogec-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Event Slot empty should render an sd slot 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1arzi5p-MuiSvgIcon-root"
        data-testid="SdIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-6 6h4c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1h-4V9zm-3.5 4.5v-1H7c-.55 0-1-.45-1-1V10c0-.55.45-1 1-1h3c.55 0 1 .45 1 1v1H9.5v-.5h-2v1H10c.55 0 1 .45 1 1V14c0 .55-.45 1-1 1H7c-.55 0-1-.45-1-1v-1h1.5v.5h2zm5 0h2v-3h-2v3z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1fyogec-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Event Slot not empty should remove the assigned event 1`] = `
HTMLCollection [
  <div
    class="empity-slot css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1fyogec-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Event Slot not empty should render 1`] = `
HTMLCollection [
  <div
    class="css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-gk29l9-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root  css-14eav6w"
        data-testid="warningIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1j8dm2n-MuiSvgIcon-root"
          data-testid="ImageNotSupportedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap  css-14y2rlv-MuiTypography-root"
        data-test="event-title"
        title="event-title"
      >
        event-title
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-1wr1quj-MuiSvgIcon-root"
        data-testid="EventAvailableIcon"
        focusable="false"
        title="Start Program"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.53 11.06 15.47 10l-4.88 4.88-2.12-2.12-1.06 1.06L10.59 17l5.94-5.94zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-startProgram"
      >
        24.12.2021 13:00
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-hkadye-MuiSvgIcon-root"
        data-testid="EventBusyIcon"
        focusable="false"
        title="End Program"
        viewBox="0 0 24 24"
      >
        <path
          d="m9.31 17 2.44-2.44L14.19 17l1.06-1.06-2.44-2.44 2.44-2.44L14.19 10l-2.44 2.44L9.31 10l-1.06 1.06 2.44 2.44-2.44 2.44L9.31 17zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-endProgram"
      >
        24.12.2022 13:00
      </p>
    </div>
  </div>,
]
`;

exports[`Event Slot not empty should render a disabled slot 1`] = `
HTMLCollection [
  <div
    class="css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root  css-14eav6w"
        data-testid="warningIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1j8dm2n-MuiSvgIcon-root"
          data-testid="ImageNotSupportedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap  css-14y2rlv-MuiTypography-root"
        data-test="event-title"
        title="event-title"
      >
        event-title
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-1wr1quj-MuiSvgIcon-root"
        data-testid="EventAvailableIcon"
        focusable="false"
        title="Start Program"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.53 11.06 15.47 10l-4.88 4.88-2.12-2.12-1.06 1.06L10.59 17l5.94-5.94zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-startProgram"
      >
        24.12.2021 13:00
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-hkadye-MuiSvgIcon-root"
        data-testid="EventBusyIcon"
        focusable="false"
        title="End Program"
        viewBox="0 0 24 24"
      >
        <path
          d="m9.31 17 2.44-2.44L14.19 17l1.06-1.06-2.44-2.44 2.44-2.44L14.19 10l-2.44 2.44L9.31 10l-1.06 1.06 2.44 2.44-2.44 2.44L9.31 17zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-endProgram"
      >
        24.12.2022 13:00
      </p>
    </div>
  </div>,
]
`;

exports[`Event Slot not empty should render an hd slot 1`] = `
HTMLCollection [
  <div
    class="css-1fk8khu"
  >
    <div
      class="MuiBox-root css-1smxw4b"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root css-1kosrk6"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-8 12H9.5v-2h-2v2H6V9h1.5v2.5h2V9H11v6zm2-6h4c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1h-4V9zm1.5 4.5h2v-3h-2v3z"
        />
      </svg>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-gk29l9-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root  css-14eav6w"
        data-testid="warningIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
      <div
        class="MuiBox-root css-1vwnyhv"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1j8dm2n-MuiSvgIcon-root"
          data-testid="ImageNotSupportedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap  css-14y2rlv-MuiTypography-root"
        data-test="event-title"
        title="event-title"
      >
        event-title
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-1wr1quj-MuiSvgIcon-root"
        data-testid="EventAvailableIcon"
        focusable="false"
        title="Start Program"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.53 11.06 15.47 10l-4.88 4.88-2.12-2.12-1.06 1.06L10.59 17l5.94-5.94zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-startProgram"
      >
        24.12.2021 13:00
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-19sicoy-MuiGrid-root"
      justify="flex-end"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium  css-hkadye-MuiSvgIcon-root"
        data-testid="EventBusyIcon"
        focusable="false"
        title="End Program"
        viewBox="0 0 24 24"
      >
        <path
          d="m9.31 17 2.44-2.44L14.19 17l1.06-1.06-2.44-2.44 2.44-2.44L14.19 10l-2.44 2.44L9.31 10l-1.06 1.06 2.44 2.44-2.44 2.44L9.31 17zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-xwvvpn-MuiTypography-root"
        data-test="event-endProgram"
      >
        24.12.2022 13:00
      </p>
    </div>
  </div>,
]
`;
